src/Compilation.zig

```diff
@@ -375,6 +381,8 @@ fn generateBuiltinTypes(comp: *Compilation) !void {
 
     const ptrdiff = if (os == .windows and comp.target.cpu.arch.ptrBitWidth() == 64)
         Type{ .specifier = .long_long }
+    else if (comp.target.cpu.arch == .msp430)
+        Type{ .specifier = .int }
```

Comment:
Wouldn't a more general solution be adding a prong for 16 bit architectures? I'd assume most of those also define `int` to be 16 bits.